src/features/surveys/models/SurveyDataModel.spec.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  ZetkinSurveyExtended,
+  ZetkinSurveySubmission,
+} from 'utils/types/zetkin';
+
+function mockList<DataType extends { id: number }>(items?: DataType[]) {
+  const list = remoteList<DataType>(items);
+  if (items) {
+    list.items = items.map((data) => mockItem(data));
+  }
+
+  return list;
+}
+
+function mockItem<DataType extends { id: number }>(data: DataType) {
+  const item = remoteItem<DataType>(data.id, { data });
+  item.loaded = new Date().toISOString();
+  return item;
+}
+describe('SurveyDataModel', () => {
+  const mockClient = mock<IApiClient>();
+  const mockRouter = mock<NextRouter>();
+
+  beforeEach(() => {
+    reset(mockClient);
+    reset(mockRouter);
+  });
+
+  describe('state', () => {
+    const mockStoreData = (
+      published: string | null,
+      expires: string | null
+    ) => ({
+      surveys: {
+        submissionList: mockList<ZetkinSurveySubmission>([]),
+        surveyList: mockList<ZetkinSurveyExtended>([
+          {
+            access: 'open',
+            allow_anonymous: true,
+            callers_only: false,
+            elements: [],
+            expires: expires,
+            id: 1,
+            info_text: 'Semla',
+            organization: {
+              id: 1,
+              title: 'Semla lovers',
+            },
+            published: published,
+            title: 'Semla lovers assemble',
+          },
+        ]),
+      },
+    });
+
+    it('is SCHEDULED when publish date is in future', () => {
+      const mockData = mockStoreData('2022-03-23', '2022-04-01');
+      jest.useFakeTimers().setSystemTime(new Date('2022-03-01'));
+      const store = createStore(mockData);
+
+      const apiClient = instance(mockClient);
+      const env = new Environment(store, apiClient, instance(mockRouter));
+      const model = new SurveyDataModel(env, 1, 1);
+      expect(model.state).toBe(SurveyState.SCHEDULED);
+    });
+
+    it('is DRAFT when publish date and expire date are null', () => {
+      const mockData = mockStoreData(null, null);
+      jest.useFakeTimers().setSystemTime(new Date('2022-03-01'));
+      const store = createStore(mockData);
+
+      const apiClient = instance(mockClient);
+      const env = new Environment(store, apiClient, instance(mockRouter));
+      const model = new SurveyDataModel(env, 1, 1);
+      expect(model.state).toBe(SurveyState.DRAFT);
+    });
+
+    it('is UNPUBLISHED when publish date and expire date are in the past', () => {
+      jest.useFakeTimers().setSystemTime(new Date('2022-04-10'));
+      const mockData = mockStoreData('2022-03-23', '2022-04-01');
+      const store = createStore(mockData);
+
+      const apiClient = instance(mockClient);
+      const env = new Environment(store, apiClient, instance(mockRouter));
+      const model = new SurveyDataModel(env, 1, 1);
+      expect(model.state).toBe(SurveyState.UNPUBLISHED);
+    });
+
+    it('is PUBLISHED when publish date is past and expireDate is in the future', () => {
```

Comment:
Minor: Perhaps `expireDate` OR `expire date` (as in the previous test `it` description).

Reply:
Thanks! I will fix it now!